check.py

```diff
@@ -522,30 +523,32 @@ def run_unittest():
     raise Exception("unittest failed")
 
 
+TEST_SUITES = OrderedDict([
+  ('help', run_help_tests),
+  ('wasm-opt', run_wasm_opt_tests),
+  ('asm2wasm', asm2wasm.test_asm2wasm),
+  ('asm2wasm-binary', asm2wasm.test_asm2wasm_binary),
+  ('wasm-dis', run_wasm_dis_tests),
+  ('crash', run_crash_tests),
+  ('dylink', run_dylink_tests),
+  ('ctor-eval', run_ctor_eval_tests),
+  ('wasm-metadce', run_wasm_metadce_tests),
+  ('wasm-reduce', lambda: run_wasm_reduce_tests() if has_shell_timeout() else None),
+  ('spec', run_spec_tests),
+  ('binaryenjs', binaryenjs.test_binaryen_js),
+  ('lld', lld.test_wasm_emscripten_finalize),
+  ('wasm2js', wasm2js.test_wasm2js),
+  ('validator', run_validator_tests),
+  ('vanilla', lambda: run_vanilla_tests() if has_vanilla_emcc and has_vanilla_llvm and 0 else None),
```

Comment:
Would it make sense to move these predicates into the functions themselves as early returns?

Reply:
Done!